src/codesyncd/handlers/tabs_handler.ts

```diff
@@ -0,0 +1,169 @@
+import path from "path";
+
+import { glob } from "glob";
+
+import { ITabYML } from "../../interface";
+import { generateSettings } from "../../settings";
+import { readYML } from "../../utils/common";
+import { getRandomIndex, getTabsBeingProcessed } from "../utils";
+import { removeFile } from "../../utils/file_utils";
+import { TAB_FILES_PER_ITERATION, TAB_SIZE_LIMIT } from "../../constants";
+import { CodeSyncLogger } from "../../logger";
+import { TabValidator } from "../validators/tab_validator";
+import { TabHandler } from "./tab_handler";
+import { ConfigUtils } from "../../utils/config_utils";
+
+export class TabsHandler {
+    // @ts-ignore
+    tabs: ITabYML[];
+    // @ts-ignore
+    accessToken: string;
+
+    settings: any;
+    configJSON: any;
+    configRepo: any;
+
+    constructor(repoTab: ITabYML[] | null = null, accessToken: string | null = null) {
+        console.log(`settins: ${this.settings}`);
+        // @ts-ignore
+        this.accessToken = accessToken;
+        // @ts-ignore
+        this.tabs = repoTab;
+        this.settings = generateSettings();
+        this.configJSON = readYML(this.settings.CONFIG_PATH);
+    }
+
+    async run() {
+        const validTabs: ITabYML[] = [];
+        let tabsSize = 0;
+        for (const tab of this.tabs) {
+            const tab_handler = new TabHandler(tab, null, this.accessToken);
+            const tabToSend = await tab_handler.createTabToSend();
+            if (!tabToSend) {
+                CodeSyncLogger.error(`createTabToSend() returned empty response`);
+                return;
+            }
+            console.log(`tabToSend: ${JSON.stringify(tabToSend)}`);
+           
+            console.log(`tabsSize: ${tabsSize}`)
+            tabsSize += JSON.stringify(tabToSend).length;
+            console.log(`tabsSize: ${tabsSize}`)
+            console.log(`validTabs: ${validTabs}`);
+            if (tabsSize < TAB_SIZE_LIMIT) {
+                validTabs.push(tabToSend);
+            } else {
+                CodeSyncLogger.error(`Tabs size limit reached, size = ${tabsSize} bytes`);
+            }
+            console.log(`validTabs: ${JSON.stringify(validTabs)}`);    
+        }
+
+        return validTabs;
+    }
+
+    getYMLFiles = async () => {
+        const tabsBeingProcessed = getTabsBeingProcessed();
+        console.log(`tabs_path: ${this.settings.TABS_PATH}`);
+        // Discard all files that aren't of .YML 
+        const invalidTabFiles = await glob("**", { 
+			ignore: "*.yml",
+			nodir: true,
+			dot: true,
+            cwd: this.settings.TABS_PATH
+        });
+
+        invalidTabFiles.forEach(invalidTabFile => {
+            const filePath = path.join(this.settings.TABS_PATH, invalidTabFile);
+            removeFile(filePath, "cleaningInvalidTabFiles");
+        })
+
+        // Get valid files
+        const tabs = await glob("*.yml", { 
+            cwd: this.settings.TABS_PATH,
+			maxDepth: 1,
+			nodir: false,
```

Comment:
What is the default value of these params?

Reply:
I believe they are undefined by default